include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl.h

```diff
@@ -310,7 +351,10 @@ struct __parallel_scan_submitter<_CustomName, __internal::__optional_kernel_name
         auto __max_cu = oneapi::dpl::__internal::__max_compute_units(__exec);
         // get the work group size adjusted to the local memory limit
         // TODO: find a way to generalize getting of reliable work-group sizes
-        ::std::size_t __wgroup_size = oneapi::dpl::__internal::__slm_adjusted_work_group_size(__exec, sizeof(_Type));
+        ::std::size_t __wgroup_size_max =
+            oneapi::dpl::__internal::__slm_adjusted_work_group_size(__exec, sizeof(_Type));
+
+        ::std::size_t __wgroup_size = ::std::min(__wgroup_size_max, __preferred_wgroup_size(_AlgoType::value));
```

Comment:
May we pack this call ```std::min``` into ```__preferred_wgroup_size``` function and pass ```__wgroup_size_max``` into them?

Reply:
We could, but we would have to modify the switch statement. Instead of returning the number, we would update a local variable and them at the end of the function, we would return the min of that variable and the maximum work-group size. I do not know if that is cleaner.